### ISSUE TITLE
"bus error" in k8s 

### ISSUE THREAD

**Issue by MichaelSp** [nextcloud/docker#1976]
After updating to 
 * **kubernetes** 1.26.4
 * **nextcloud** 26.0.1-apache

I had to set  `opcache.enable=0` the following line in https://github.com/nextcloud/docker/blob/2d8f990304e76071602375a197b65692fa2de69a/24/apache/Dockerfile#L95

If I set it to `1` the container is in crash-loop with this error message:

```
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 2001:cafe:42::249f. Set the 'ServerName' directive globally to suppress this message
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 2001:cafe:42::249f. Set the 'ServerName' directive globally to suppress this message
Bus error
```

btw: I'm using a [fork](https://github.com/MichaelSp/nextcloud-helm) of https://github.com/nextcloud/helm

**Comment by joshtrichards** [comment]
Are you asking for something here or just getting this noted in case someone has the same issue and searches for it?

This seems to either be a Kubernetes or an upstream matter, no? Or, from the looks of your fork patch, perhaps something applicable to the nc-helm project, but not here.

Disabling opcache for everyone else using this Docker image is almost certainly a nonstarter. :)

**Comment by MichaelSp** [comment]
I completely agree. Disabling `opchache` is not a good solution. I was not recommending that. I'd consider this a bug report and wanted to see if someone has the same issue and has a better solution.

Meanwhile I also figured out that I had very high CPU-usage (I was hacked and there was a crypto-miner running). But I would still consider this a bug, because the docker container should start properly even under high system load.

If you (or someone else? Maintainer?) don't want to follow up on this, feel free to close this issue. Having that documented for posterity is already something.

**Comment by joshtrichards** [comment]
Related it seems: #1859

**Comment by trunet** [comment]
I'm having this on the fpm image.

I strace php-fpm and got:

```
open("/tmp/.ZendSem.BLocAD", O_RDWR|O_CREAT|O_EXCL|O_LARGEFILE, 0600) = 3
fchmod(3, 0666)                         = 0
fcntl(3, F_GETFD)                       = 0
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
unlink("/tmp/.ZendSem.BLocAD")          = 0
open("/proc/self/maps", O_RDONLY|O_LARGEFILE) = 4
read(4, "55a77b400000-55a77b513000 r--p 0"..., 1024) = 1024
lseek(4, -281, SEEK_CUR)                = 743
close(4)                                = 0
mmap(0x55a77d800000, 268435456, PROT_READ|PROT_WRITE, MAP_SHARED|MAP_FIXED|MAP_ANONYMOUS|MAP_HUGETLB, -1, 0) = 0x55a77d800000
--- SIGBUS {si_signo=SIGBUS, si_code=BUS_ADRERR, si_addr=0x55a77d800000} ---
+++ killed by SIGBUS +++
Bus error
```

php-fpm (opcache in reality) also uses HUGETLB, the helm chart documentation needs to be updated to mention fpm image as well instead of only apache version. both fpm and fpm-alpine. If I disable opcache, it works fine.

However the best alternative is to enable hugepages on fpm as well. https://github.com/nextcloud/helm/tree/main/charts/nextcloud#hugepages

**Comment by joshtrichards** [comment]
Sounds like this is a docs matter for the Helm chart then? I'll move this there since doesn't seem like anything in the image itself.
